src/percy-agent-client/percy-agent.ts

```diff
@@ -6,41 +6,52 @@ export default class PercyAgent {
   clientInfo: string | null
   environmentInfo: string | null
   xhr: any
+  postSnapshotDirectly: boolean
   port: number
   domTransformation: any | null
-  client: PercyAgentClient
+  client: PercyAgentClient | null
```

Comment:
This is something for discussion: If we don't want `PercyAgent` to be posting directly, then it doesn't need the `xhr` and it doesn't need the `client`. Do we still create and set those fields, or not?

Right now I've chosen to go with the path where, if we're not going to be posting directly, we leave those null. An advantage of this is that now you can instantiate `PercyAgent` from an environment where there isn't an `XMLHTTPRequest` available. The disadvantage is that now we have to check if those fields are null, and it is not the most elegant thing.

Reply:
I think that's a good tradeoff. Seems reasonable to me to avoid creating objects that won't be used.